website/blog/2023-08-08-deploy-run-ferretdb-leafcloud.md

```diff
@@ -0,0 +1,443 @@
+---
+slug: deploy-run-ferretdb-leafcloud
+title: 'How to Deploy and Run FerretDB on Leafcloud'
+authors: [alex]
+description: >
+  In this how-to guide, we’ll be showing you how to configure, deploy, and run FerretDB on [Leafcloud](https://www.leaf.cloud/), using the power of Kubernetes and the PostgreSQL Operator.
+image: /img/blog/ferretdb-leafcloud.jpg
+tags: [compatible applications, tutorial, cloud]
+---
+
+![How to Deploy and Run FerretDB on leafcloud](/img/blog/ferretdb-leafcloud.jpg)
+
+In this how-to guide, we'll be showing you how to configure, deploy, and run FerretDB on [Leafcloud](https://www.leaf.cloud/), using the power of Kubernetes and the PostgreSQL Operator.
+
+<!--truncate-->
+
+Building resilient, scalable, and performant applications today can be quite a complex process.
+As such, [FerretDB](https://www.ferretdb.io), an open-source database alternative to MongoDB that leverages PostgreSQL as the backend, can be a strategic choice for users looking to avoid vendor lock-in and build a totally flexible and reliable backend for many applications.
+
+On the other hand, Leafcloud is an eco-friendly cloud infrastructure that provides a distributed architecture for you to run your FerretDB application.
+
+This guide will offer you a comprehensive, step-by-step guide to navigating the intricacies of setting up a Kubernetes containerization environment, persistent storage provisioning, and FerretDB deployment on Leafcloud.
+
+Let's get started!
+
+## Setting up FerretDB on top of Leafcloud
+
+We're going to set up FerretDB on a Kubernetes cluster and deploy it to Leafcloud.
+
+### Prerequisites
+
+- Kubectl
+- Leafcloud account (ensure you have enough volumes to run the cluster)
+
+### Creating a Kubernetes Cluster in Leafcloud
+
+We will start by creating a Kubernetes cluster, using the OpenStack CLI.
+Leafcloud manages and deploys container clusters using the OpenStack Magnum project.
+
+Start by installing the OpenStack CLI using the following command:
+
+```sh
+sudo apt update -y
+sudo apt install -y python3-pip python3-dev -y
+sudo apt install virtualenv -y
+virtualenv -p python3 openstack_venv
+source openstack_venv/bin/activate
+pip install --upgrade pip
+pip install python-openstackclient
+pip install python-magnumclient
+```
+
+Once it's installed, you need to log in to your Leafcloud account to download the OpenStack RC file that contains the environment variables for your command-line client.
```

Comment:
Again would be nice to refer to Leafcloud docs in addition (for other OS, or updates, or something)